compiler/semexprs.nim

```diff
@@ -643,6 +643,15 @@ proc arrayConstrType(c: PContext, n: PNode): PType =
   result = typ
 
 proc semArrayConstr(c: PContext, n: PNode, flags: TExprFlags; expectedType: PType = nil): PNode =
+  if n.typ != nil and n.typ.kind == tySequence:
```

Comment:
Sorry if this is outside your comfort range, but could we maybe wait until the end and sem the array as normal and finally change the type to `seq` from the formed `array` type if the node was originally typed as `seq`?